### PR TITLE
[BUILD] Pick Commons-Logging dependence out of Hudi-Common

### PR DESCRIPTION
**Pull request by zwangsheng**
### _Why are the changes needed?_

```
        <dependency>
            <groupId>org.apache.hadoop</groupId>
            <artifactId>hadoop-client-minicluster</artifactId>
            <scope>test</scope>
        </dependency>
```
`hadoop-client-minicluster` need `commons-logging` dependence. Now it works because `hudi-spark-common` include this dependency.

We extract it to better manage dependencies.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by pan3793** [comment]
`org.slf4j:jcl-over-slf4j` is a drop-in replacement of `commons-logging`, please use the former.

**Comment by ulysses-you** [comment]
+1 with @pan3793

**Comment by zwangsheng** [comment]
@pan3793 I check https://www.slf4j.org/faq.html#excludingJCL found that provides two methods to exclude commons - logging, I prefer to the second, what do you think?

**Comment by pan3793** [comment]
> @pan3793 I check https://www.slf4j.org/faq.html#excludingJCL found that provides two methods to exclude commons - logging, I prefer to the second, what do you think?

Your change does not implement option 2, it still in test scope.
Personally I prefer option 1, because we don't need to declare versions for un-used deps.

**Comment by zwangsheng** [comment]
@pan3793 I do as you said and found that this class  was missing org.apache.commons.logging.impl.Log4JLogger

**Comment by pan3793** [comment]
I see what happens, it access the `Log4JLogger` directly, let's remove the exclusion from the `hadoop-client-runtime` in root pom

File: kyuubi-server/pom.xml [hunk 1]
```diff
@@ -349,6 +349,12 @@
             <scope>test</scope>
         </dependency>
 
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>jcl-over-slf4j</artifactId>
```
Comment (pan3793) [inline kyuubi-server/pom.xml:354]: Then we don't need it.

File: pom.xml [hunk 2]
```diff
@@ -180,7 +180,8 @@
         <maven.plugin.scala.version>4.3.0</maven.plugin.scala.version>
         <maven.plugin.surefire.version>2.22.0</maven.plugin.surefire.version>
         <maven.plugin.scalatest.version>2.0.2</maven.plugin.scalatest.version>
-        <maven.plugin.scalatest.exclude.tags>org.apache.kyuubi.tags.ExtendedSQLTest,org.apache.kyuubi.tags.HudiTest</maven.plugin.scalatest.exclude.tags>
+        <maven.plugin.scalatest.exclude.tags>org.apache.kyuubi.tags.ExtendedSQLTest,org.apache.kyuubi.tags.HudiTest
+        </maven.plugin.scalatest.exclude.tags>
```
Comment (pan3793) [inline pom.xml:184]: Unnecessary chagnes?

Comment (zwangsheng) [reply pom.xml:184]: `./build/dependency.sh --replace` changed those.

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging.

**Pull request closed without merging**